### ISSUE TITLE
azure_rm_virtualmachine VM start and stop

### ISSUE THREAD

**Issue by daica75** [VSChina/ansible-issue-tracker#950]
I am getting the following error when I try to start or stop a virtual machine, I am getting the following:

The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible-tmp-1559657705.67-271285273530579/AnsiballZ_azure_rm_virtualmachine.py", line 114, in 
    _ansiballz_main()
  File "/tmp/ansible-tmp-1559657705.67-271285273530579/AnsiballZ_azure_rm_virtualmachine.py", line 106, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/tmp/ansible-tmp-1559657705.67-271285273530579/AnsiballZ_azure_rm_virtualmachine.py", line 49, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py", line 2045, in 
  File "/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py", line 2041, in main
  File "/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py", line 853, in __init__
  File "/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/ansible_azure_rm_virtualmachine_payload.zip/ansible/module_utils/azure_rm_common.py", line 325, in __init__
  File "/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py", line 1434, in exec_module
KeyError: 'diskSizeGB'
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible-tmp-1559657705.67-271285273530579/AnsiballZ_azure_rm_virtualmachine.py\", line 114, in \n    _ansiballz_main()\n  File \"/tmp/ansible-tmp-1559657705.67-271285273530579/AnsiballZ_azure_rm_virtualmachine.py\", line 106, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/tmp/ansible-tmp-1559657705.67-271285273530579/AnsiballZ_azure_rm_virtualmachine.py\", line 49, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py\", line 2045, in \n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py\", line 2041, in main\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py\", line 853, in __init__\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/ansible_azure_rm_virtualmachine_payload.zip/ansible/module_utils/azure_rm_common.py\", line 325, in __init__\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_I72j_4/__main__.py\", line 1434, in exec_module\nKeyError: 'diskSizeGB'\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", 
    "rc": 1
}

It seems to be erroring on a keyword diskSizeGB

**Comment by mmyyrroonn** [comment]
@daica75 Hello. Can you share the playbook file(.yml) and the Ansible version information with us? BTW, If you can open the issue in the main [repo](https://github.com/ansible/ansible), it will be much helpful and we can better track the issue. 😄

**Comment by mmyyrroonn** [comment]
@daica75 Hello. I checked the code. I'm not sure if it will solve your problem. The parameter used for setting the OS disk size in GB is called `os_disk_size_gb`. I hope it can solve your question. 😄

**Comment by Fred-sun** [comment]
@daica75 Are you still here? Could you can share more info for this issue? How to reproduce, ansible version, how do you create the VM ....., also need you help provide the VM's facts, try this? Thank you very much!
```e.g.
    - name: Get Virtualmachine facts
      azure_rm_virtualmachine_facts:
        resource_group: "{{ resource_group }}"
        name: "{{ vm_name }}"

**Comment by daica75** [comment]
Hi, I am sorry I forgot about this.

What info do you need to try to replicate this?

I got around this issue by having the Azure VM be turned on manually through the console first, then after that when I run the processes through Ansible, I am able to turn off the VM, and from then I am able to control it to turn on or turn off.

The main part of my playbook doing the Azure VM control is just doing the generic call

azure_rm_virtualmachine:
  resource_group: <my Azure resource group>
  name: <name of Azure VM>
  started: yes
  allocated: yes

or 

azure_rm_virtualmachine:
  resource_group: <my Azure resource group>
  name: <name of Azure VM>
  started: no
  allocated: no

**Comment by daica75** [comment]
Hi

Here is the azure_rm_virtualmachine_facts from a machine that is not working:
ok: [localhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "name": "azcmacc05",
            "password": null,
            "profile": null,
            "resource_group": "myapp_SEASIA",
            "secret": null,
            "subscription_id": null,
            "tags": null,
            "tenant": null
        }
    },
    "vms": [
        {
            "admin_username": "ccbuild-temp",
            "data_disks": [
                {
                    "caching": "ReadOnly",
                    "disk_size_gb": null,
                    "lun": 0,
                    "managed_disk_type": null
                }
            ],
            "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myapp_SEASIA/providers/Microsoft.Compute/virtualMachines/azcmacc05",
            "image": {
                "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myapp_SEAsia/providers/Microsoft.Compute/images/volata-accelerator-image1"
            },
            "location": "southeastasia",
            "name": "azcmacc05",
            "network_interface_names": [
                "azcmacc05833"
            ],
            "os_disk_caching": "ReadWrite",
            "os_type": "Windows",
            "power_state": "deallocated",
            "resource_group": "myapp_SEASIA",
            "state": "present",
            "tags": {
                "Application": "myapp"
            },
            "vm_size": "Standard_F8s_v2"
        }
    ]
}

I did the same thing for a machine that is working:
ok: [localhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "auth_source": null,
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "name": "azvmptbg",
            "password": null,
            "profile": null,
            "resource_group": "myapp_SEASIA",
            "secret": null,
            "subscription_id": null,
            "tags": null,
            "tenant": null
        }
    },
    "vms": [
        {
            "admin_username": "ccbuild-temp",
            "data_disks": [
                {
                    "caching": "ReadOnly",
                    "disk_size_gb": null,
                    "lun": 0,
                    "managed_disk_type": null
                }
            ],
            "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myapp_SEASIA/providers/Microsoft.Compute/virtualMachines/azvmptbg",
            "image": {
                "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myapp_SEAsia/providers/Microsoft.Compute/images/portbridge-win7build-image"
            },
            "location": "southeastasia",
            "name": "azvmptbg",
            "network_interface_names": [
                "azvmptbg421"
            ],
            "os_disk_caching": "ReadOnly",
            "os_type": "Windows",
            "power_state": "deallocated",
            "resource_group": "myapp_SEASIA",
            "state": "present",
            "tags": {
                "Application": "myapp_Build"
            },
            "vm_size": "Standard_F8s_v2"
        }
    ]
}


Basically, to fix the issue, what I found I had to do is that I just have the machine on first through Azure, then after that I would be able to turn it off with the Ansible module, and from then I would be able to start up and shut down the Azure VM with the Ansible module.

Also, I am using Ansible 2.8.0 right now
-bash-4.2# ansible --version
ansible 2.8.0
  config file = None
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
-bash-4.2#

**Comment by Fred-sun** [comment]
@daica75 Thanks for your update, I try but can't reproduce. others, If you can open the issue in the main [repo](https://github.com/ansible/ansible), it will be much helpful and we can better track the issue.

**Comment by Fred-sun** [comment]
@daica75 Try and copy the error. It cannot be copied. Can you copy the problem now? Thank you!

**Comment by Fred-sun** [comment]
@MyronFanQiu Please help take a look at this issue when you're free! Thank you!

**Comment by Fred-sun** [comment]
@daica75  Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information.
Please set the description of this issue with this template:
https://github.com/ansible/ansible/blob/devel/.github/ISSUE_TEMPLATE/bug_report.md
if I get the reproduce method , I will try and push issue effectively to solve.E25

**Comment by Fred-sun** [comment]
@daica75 I can never copy your problem, can you update to the latest version and try again? Is there this Issue? Thanks  a lot!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@daica75 Would you please retry it in latest ansible! Thank you very much!

**Comment by daica75** [comment]
Sorry I have missed these emails before. I will plan on retrying with new Ansible version.

Sent from my iPhone

On Oct 12, 2019, at 1:53 AM, Fred-sun <notifications@github.com> wrote:

﻿

@daica75<https://github.com/daica75> Would you please retry it in latest ansible! Thank you very much!

**Comment by Fred-sun** [comment]
@daica75 Can you update the test results? Does the new version of Ansible solve your problem? Thank you very much!

**Comment by Fred-sun** [comment]
> Sorry I have missed these emails before. I will plan on retrying with new Ansible version.
> …

@daica75 Any update for the result? I want to know more about this!. Thank you very much!

**Comment by Fred-sun** [comment]
@daica75 Kindly ping!
